lessons/Data-Science/contracts-tables-visualizations/langs/en-us/pages/functions-for-tables.adoc

```diff
@@ -7,70 +7,67 @@
+@n `true` sorts the table... @fitb{}{@ifsoln{from A-Z ... or from least to greatest}}
 
-@n Which animal(s) were adopted the quickest? @fitb{}{@ifsoln{Snuggles, Nori, Toggle, Wade and Mittens}}
+@n `false` sorts the table... @fitb{}{@ifsoln{from Z-A ... or from greatest to least}}
 
-@n Some functions produce Numbers, some produce Strings, some produce Booleans. What did the `sort` function produce? @fitb{}{@ifsoln{a new table!}}
+@n What code will sort the animals by alphabetical order of their _names_? @fitb{}{@ifsoln{@show{(code '(sort animals-table "name" true))}}}
 
-@vspace{1ex}
+@n Did you use `true` or `false`? Explain why. @fitb{}{}
 
-There are many other functions available to us in Pyret. We can describe them using contracts. The Contract for `sort` is:
+== Summarizing by a Column
+Let's explore the contract for another table function:
 
-@indented{@show{(contract "sort" '("Table" "String" "Boolean") "Table")}}
+@show{(contract "count" '("Table" "String") "Table")}
 
-- Each Contract begins with the function name: @fitb{10em}{_in this case_ `sort`}
-- Lists the data types required to satisfy its Domain: @fitb{20em}{_in this case_ Table, String, Boolean}
-- And then declares the data type of the Range it will return. @fitb{10em}{_in this case_ Table}
-- Contracts can also be written with more detail, by adding _variable names_ in the Domain:
+@vspace{1ex}
 
-@indented{@show{(contract 'sort '((table-name Table) (column-name String) (order Boolean)) "Table")}}
+*Suppose we wanted to know how many animals had 4 legs...*
 
-@vspace{1ex}
+Type @show{(code '(count animals-table "legs"))} into the Interactions Area and click "Enter"
+
+@n What did the expression produce? @fitb{}{@ifsoln{a new table showing the counts for the number of animals with 0 legs, 3 legs, 4 legs and 8 legs}}
 
-*Suppose we wanted to sort the `animals-table` by the `legs` column to determine which animals had the most legs...*
+@n How many animals had 4 legs? @fitb{}{@ifsoln{29}}
 
-@n Fill in the blanks below with the code you'd use (We've put pieces of the Contract below each line to help you!):
+@n Think of another question you might be able to answer with the `count` function.
 
-@fitbruby{10em}{@ifsoln{sort}}{function-name} ( @fitbruby{20em}{@ifsoln{animals-table}}{table-name {two-colons} Table},  @fitbruby{15em}{@ifsoln{"legs"}}{column-name {two-colons} String}, @fitbruby{10em}{@ifsoln{true}}{order {two-colons} Boolean})
+@fitb{}{@ifsoln{Possible response: How many animals are cats?}}
 
-@n Which animal(s) had the most legs? @fitb{}{@ifsoln{Snuggles the tarantula}}
+@n Fill in the blanks with the code you'd write.
 
-@n Think of another question you might answer quickly by sorting the table.
+@fitbruby{10em}{@ifsoln{count}}{function-name} (@fitbruby{20em}{@ifsoln{animals-table}}{table-name {two-colons} Table}, @fitbruby{15em}{@ifsoln{Answers will vary!}}{column-name {two-colons} String})
```

Comment:
@schanzer The rendering of this is inconsistent. Sometimes it works. But most of the time I see "two-colons" and the datatype appears on the fitb.